### ISSUE TITLE
Deserialize arrays containing objects fails

### ISSUE THREAD

**Issue by Yvi71** [danieleteti/delphimvcframework#404]
while deserializing Json objects containing records containing sub-objects, i encountered a problem. 
Maybe this is quite uncommon or i did something wrong with the usage of the serializer, but i could provide code to fix the issue. It will deserialize such objects in gerneric Lists (TList<Objecttype>) using rtti info.  Therefore i extended the TMVCJsonDataObjectsSerializer.JsonDataValueToAttribute method for the case the target type is a generic list. Is that of interest?

**Comment by danieleteti** [comment]
Yes, create a sample project with a couple of tests (serialization and deserialization with border case too) and we'll manage to merge into the repo.

**Comment by danieleteti** [comment]
This issue should be fixed now, can you test it?
